### DISCUSSION TITLE
Unsupported Pattern: Importing Server Components into Client Components

### DISCUSSION THREAD

**Discussion by d1d4r** [vercel/next.js#65461] [Help]
### Summary

hi there,

nextjs docs say you cant import server component into server component, but i do 

[https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#unsupported-pattern-importing-server-components-into-client-components](url)

no error no warning it just work well

todo.jsx is server component
```
import axios from "axios";
import React from "react";

export default async function Todo({ passProps, pathname }) {
  const { data } = await axios.get("https://dummyjson.com/todos");

  return (
    <div>
      <p>
        {passProps}:{pathname}
      </p>
      <ul>
        {data.todos.map((todo) => {
          return <li key={todo.id}>{JSON.stringify(todo)}</li>;
        })}
      </ul>
    </div>
  );
}
```
client-todo.jsx is client component

```
 "use client";
import React from "react";
import Todo from "./Todo";
import { usePathname } from "next/navigation";

export default function ClientTodo({ testProps }) {
  const pathname = usePathname();
  return (
    <div>
      <p>{testProps}</p>
      <Todo passProps="from client" pathname={pathname} />
    </div>
  );
}
```
as u see, i can also pass props ?!!

**Comment by icyJoseph** [comment]
Hi,

Now try to put a `useState` hook in `ClientTodo`, and update that state, like:

```tsx
'use client';
import React, { useState } from 'react';
import Todo from './Todo';
import { usePathname } from 'next/navigation';

export default function ClientTodo({ testProps }) {
  const pathname = usePathname();
  const [value, setValue] = useState(0);
  return (
    <div>
      <p>{value}</p>
      <button onClick={() => setValue(100)}>click</button>
      <Todo passProps="from client" pathname={pathname} />
    </div>
  );
}

```

**Reply by d1d4r** [reply]
Same thing bro 😅 no error no warning 

But state can't re-render dom

There is no different between usePathname and useState both of them are hook,
Hook not allowed, it should show error

**Reply by icyJoseph** [reply]
it will error, it crashes with something like, `Client components can't async/await`

**Reply by icyJoseph** [reply]
<img width="1212" alt="Screenshot 2024-05-07 at 17 46 48" src="https://github.com/vercel/next.js/assets/21013447/96956d93-7908-41b9-adf0-4f5376b44d30">

**Reply by d1d4r** [reply]
Yes ur right, but it didn't show error message, I was encountered those error before

**Reply by d1d4r** [reply]
Now it show, after 6 minutes ?!!? 
![image](https://github.com/vercel/next.js/assets/68716408/b6fa2123-7a93-4484-a631-88089394610c)

**Reply by d1d4r** [reply]
in Chrome didn't work, i tried in firefox, 
But aslo firefox after 6 minutes it show

**Reply by revskill10** [reply]
No, your server component will be re-render on client, it'll become client component.

**Reply by d1d4r** [reply]
It will be client if u do not use 
async/await
